bowtie/utilclasses.js

```diff
@@ -0,0 +1,443 @@
+    return Object.values(this.counts).reduce(
+      (sum, count) => sum + count.errored_tests,
+      0,
+    );
+  }
+
+  get skipped_tests() {
+    return Object.values(this.counts).reduce(
+      (sum, count) => sum + count.skipped_tests,
+      0,
+    );
+  }
+
+  // add_case_metadata(seq, caseMetadata)
+  add_case_metadata(caseObject) {
+    const results = caseObject.case.tests.map((test) => [test, {}]);
+    this._combined[caseObject.seq] = {
+      case: caseObject.case,
+      results: results,
+    };
+  }
+
+  // see_error(implementation, seq, context, caught)
+  see_error(implementationObject) {
+    const count = this.counts[implementationObject.implementation];
+    count.total_cases += 1;
+    count.errored_cases += 1;
+
+    const caseMetadata = this._combined[implementationObject.seq].case;
+    count.total_tests += caseMetadata.tests.length;
+    count.errored_tests += caseMetadata.tests.length;
+  }
+
+  see_result(result) {
+    const count = this.counts[result.implementation];
+    count.total_cases += 1;
+
+    const combined = this._combined[result.seq].results;
+
+    for (let i = 0; i < result.compare().length; i++) {
+      const [test, failed] = result.compare()[i];
+      const [, seen] = combined[i];
+      count.total_tests += 1;
+      if (test.skipped) {
+        count.skipped_tests += 1;
+        seen[result.implementation] = [test.reason, "skipped"];
+      } else if (test.errored) {
+        count.errored_tests += 1;
+        seen[result.implementation] = [test.reason, "errored"];
+      } else {
+        if (failed) {
+          count.failed_tests += 1;
+        }
+        seen[result.implementation] = [test, failed];
+      }
+    }
+  }
+
+  see_skip(skipped) {
+    const count = this.counts[skipped.implementation];
+    count.total_cases += 1;
+
+    const case_data = this._combined[skipped.seq].case;
+    count.total_tests += case_data.tests.length;
+    count.skipped_tests += case_data.tests.length;
+
+    for (const [, seen] of this._combined[skipped.seq].results) {
+      const message = skipped.issue_url || skipped.message || "skipped";
+      seen[skipped.implementation] = [message, "skipped"];
+    }
+  }
+
+  see_maybe_fail_fast(did_fail_fast) {
+    this.did_fail_fast = did_fail_fast;
+  }
+
+  case_results() {
+    return Object.values(this._combined).map((each) => ({
+      case: each.case,
+      registry: each.registry || {},
+      results: each.results,
+    }));
+  }
+
+  *flat_results() {
+    let sorted_CombinedArray = Object.entries(this._combined).sort(
+      ([keyA, valueA], [keyB, valueB]) => {
+        if (keyA < keyB) {
+          return -1;
+        } else if (keyA > keyB) {
+          return 1;
+        } else {
+          return 0;
+        }
+      },
+    );
+    let sorted_CombinedObject = Object.fromEntries(sorted_CombinedArray);
+    // console.log(sorted_CombinedObject)
+    for (let seq in sorted_CombinedObject) {
+      let each = sorted_CombinedObject[seq];
```

Comment:
Hi, I don't think this line is required here. `seq` is already the "each" in `sorted_CombinedArray` and seems to be the problem. `seq` is not a key, it's a value (in context to a key-value pair).

Reply:
No, in the raw data, `seq` is key.